api/resource/resource.go

```diff
@@ -458,10 +403,36 @@ func (r *Resource) GetNamespace() string {
 
 // OrgId returns the original, immutable ResId for the resource.
 // This doesn't have to be unique in a ResMap.
-// TODO: compute this once and save it in the resource.
 func (r *Resource) OrgId() resid.ResId {
-	return resid.NewResIdWithNamespace(
-		r.GetGvk(), r.GetOriginalName(), r.GetOriginalNs())
+	ids := r.PrevIds()
+	if len(ids) > 0 {
+		return ids[0]
+	}
+	return r.CurId()
+}
+
+// PrevIds returns a list of ResIds that includes every
+// previous ResId the resource has had through all of its
+// GVKN transformations, in the order that it had that ID.
+// The returned array does not include the resource's current
+// ID.
+// If there are no previous IDs, this will return nil.
+func (r *Resource) PrevIds() []resid.ResId {
+	var ids []resid.ResId
+
+	// names and ns should always be the same length
+	names := r.getCsvAnnotation(buildAnnotationPreviousName)
+	ns := r.getCsvAnnotation(buildAnnotationPreviousNamespace)
+	for i := range names {
+		ids = append(ids, resid.NewResIdWithNamespace(
+			r.GetGvk(), names[i], ns[i]))
+	}
+	return ids
+}
+
+// StorePreviousId stores the resource's current ID via build annotations.
+func (r *Resource) StorePreviousId() {
+	r.setPreviousNamespaceAndName(r.GetNamespace(), r.GetName())
```

Comment:
namespace is a pain, because of the confusion between an empty string, the default namespace, and cluster-scoped objects like cluster-role.  What happens to empty namespace in the CSV in the annotation?

TODO: consider  using  `(id ResId) EffectiveNamespace()` here.

That always returns an unambigous value.

Reply:
done